src/states/elder/mod.rs

```diff
@@ -1413,14 +1355,11 @@ impl Base for Elder {
         dst: DstLocation,
         content: Vec<u8>,
     ) -> Result<(), RoutingError> {
-        self.send_routing_message(
-            RoutingMessage {
-                src,
-                dst,
-                content: Variant::UserMessage(content),
-            },
-            None,
-        )
+        if let DstLocation::Direct = dst {
+            return Err(RoutingError::BadLocation);
```

Comment:
Why is this an error? if this error ever occur is it because of a bug?

Reply:
`DstLocation::Direct` is not valid in this case because `UserMessage` is a "hop" message. 

> if this error ever occur is it because of a bug?

Yes, but not in routing. This is called by the upper layer.